plugins/entity_points_converter.py

```diff
@@ -280,26 +284,28 @@ def csv_to_json_gen(file_, file_type) -> Iterator[dict]:
     # Yield all elements once in json format
     yield first_ent
     for ent in gen:
-        ent = ent._asdict()
         yield {"ID": ent["ID"], "href": ent["href"], "point": get_point(ent)}
 
 
 def json_to_json_gen(file_, file_type) -> Iterator[dict]:
     gen = load_entities(file_, mimetype=file_type)
-
     # Yield the first element twice. This will be used to later get the dimensionality of the point attribute
-    first_ent = next(gen)
-    yield first_ent
-
-    # Yield all elements once in json format
-    yield first_ent
-    for ent in gen:
-        yield ent
+    first = next(gen)
+    return itr_chain((first, first), gen)
 
 
 def get_dim_and_json_gen(file_) -> (int, Iterator[dict]):
-    file_.encoding = "utf-8"
-    input_format = file_.headers["Content-Type"]
+    header = file_.headers["Content-Type"]
+    # Header might contain more than just the input format
+    header = header.split(';')
+    input_format = header[0]
+
+    # If header contains the encoding type (charset), then take it. Otherwise, set it to utf-8
+    encoding = "utf-8"
+    for i in range(len(header)):

```

Comment:
directly iterate over the contents when the index is not required.

```python
for extra in header[1:]: # the slice ensures that the first part is ignored (e.g. the content type)
    if extra.strip().startswith("charset="):
        encoding = extra.split("=", maxsplit=1)[1].strip()
```
`strip()` gets rid of surrounding whitespace.